lib/validator/common.go

```diff
@@ -80,3 +82,39 @@ func ValidateMetadataIDsAssigned(rm unversioned.ResourceMetadata) error {
 
 	return nil
 }
+
+// ValidateIDFieldsSame is used to validate that the Resource Objects
+// supplied in one slice reference the same objects in the other slice
+// by comparing the ID fields.
+func ValidateIDFieldsSame(original, changed []unversioned.ResourceObject) error {
+	toCompare := make(map[string]int)
+	for i, obj := range original {
+		toCompare[obj.GetResourceID()] = i
+	}
+
+	for _, obj := range changed {
+		if _, ok := toCompare[obj.GetResourceID()]; !ok {
+			return fmt.Errorf("Cannot change metadata fields: %v, resource will not be found", GetIDFields(obj.GetResourceMetadata()))
+		}
+	}
+
+	return nil
+}
+
+// GetIDFields is a helper function that returns the field labels
+// on the metadata objects that are ID fields excluding Labels,
+// Annotations, and Tags
+func GetIDFields(rm unversioned.ResourceMetadata) []string {
```

Comment:
I see.  My thought was to basically write a function that would work for all the resources without needing to create more individual methods on each that would be need to be maintained should something change.  To that end, adding a new tag type would probably be best as far as that line of thinking goes.  You're right though; all of this is probably unnecessary if I can get a user friendly name working so then I can parse those values.  `WorkloadEndpoint` might have a little bit of a long string name though....